File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/models/knowledge_base_create_request.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class KnowledgeBaseCreateRequest:
    """Knowledge base creation request"""
    name: str
    description: str = None
    extra: Dict[str, Any] = None
    embedding_model_id: str = None
    index_type: str = None

File: sdkwork-ai-sdk-go/types/knowledge_base_create_request.go
package types

// Knowledge base creation request
type KnowledgeBaseCreateRequest struct {
	Name string `json:"name"`
	Description string `json:"description"`
	Extra map[string]interface{} `json:"extra"`
	EmbeddingModelId string `json:"embedding_model_id"`
	IndexType string `json:"index_type"`
}
